app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items, dependent: :destroy
+  has_many :customers, through: :customer_items
+
+  def purchase_count
+    self.customer_items.count
```

Comment:
It will produce the same output in this case, but it's generally  more precise and conventional to refer to the other entity rather than the joins record: `self.customers.count`

In this case, there's probably no difference, but if there were duplicates in the joins table, even if you tried to call `.distinct` on the query you have right now, it wouldn't remove the duplicates (since each joins record has a unique ID). If we called distinct on `self.customers.count`, the duplicate customers would be removed. 

Just something to think about for the future -- not an issue for this IC!